fix: bug in updating input & output amount

mikasackermn opened the pull request:
# Summary
solved two issues:
1. Incorrect updating of the input amount occurs when the source token is eliminated.
2. The dollar value at the destination is not updated accurately when tokens are removed.

Fixes # (issue)

The issue was that the values weren't updated appropriately when the tokens were removed.
In order to improve display, I added a condition to the inputs component that states that values will be presented correctly in the event that the token is removed or the USD value is null.

# How did you test this change?

It is preferable to take the test on the playground:

1. First, indicate the token and input's default values.
2. Navigate to the token selection page. Choose all options without choosing a token.
3. Return to the home screen.
In this instance, the USD values ought to be shown appropriately.

# Checklist:

- [x] I have performed a self-review of my code

nikaaru commented on widget/embedded/src/containers/Inputs/Inputs.tsx:
```diff
@@ -69,6 +69,11 @@ export function Inputs(props: PropTypes) {
       ? null
       : getPriceImpact(inputUsdValue.toString(), outputUsdValue.toString());
+
+  const inputUsd =
```
```suggestion
  const inputUsdAmount =
```

mikasackermn replied:
fixed

nikaaru commented on widget/embedded/src/containers/Inputs/Inputs.tsx:
```diff
@@ -69,6 +69,11 @@ export function Inputs(props: PropTypes) {
+
+  const inputUsd =
+    !inputUsdValue || !fromToken ? new BigNumber(0) : inputUsdValue;
+  const outputUsd =
```
```suggestion
  const outputUsdAmount =
```

mikasackermn replied:
fixed

mikasackermn replied:
test

nikaaru commented:
I think the value of realUsdValue should also be updated
Because now when we show zero dollar amount, the tooltip is still showing the previous amount.

mikasackermn commented:
> I think the value of realUsdValue should also be updated Because now when we show zero dollar amount, the tooltip is still showing the previous amount.
> 
> Please check if it needs to be updated elsewhere

I fixed, now updated real Usd Value when delete selected token

nikaaru commented on widget/embedded/src/containers/Inputs/Inputs.tsx:
```diff
@@ -140,11 +146,11 @@ export function Inputs(props: PropTypes) {
                 USD_VALUE_MAX_DECIMALS
               ),
-          realValue: formatTooltipNumbers(outputAmount),
+          realValue: formatTooltipNumbers(outputUsdAmount),
```
this is real output value, not real output USD value,
now, the tooltip for the real output value is incorrect,
Please revert this to its previous state.

Please make sure that all input and output items are correct for both the main value and its tooltip value and the dollar value and its tooltip value.

mikasackermn replied:
fixed

nikaaru commented on widget/embedded/src/containers/Inputs/Inputs.tsx:
```diff
@@ -55,20 +60,21 @@ export function Inputs(props: PropTypes) {
 
   const priceImpactInputCanNotBeComputed = !canComputePriceImpact(
```
the second argument of the `canComputePriceImpact` function is the input amount, not the input USD amount.

mikasackermn replied:
fixed

nikaaru commented on widget/embedded/src/containers/Inputs/Inputs.tsx:
```diff
@@ -55,20 +60,21 @@ export function Inputs(props: PropTypes) {
   const priceImpactOutputCanNotBeComputed = !canComputePriceImpact(
     selectedQuote,
-    inputAmount,
+    inputUsdAmount.toString(),
```
It should be like the input here:
the second argument of the `canComputePriceImpact` function is the input amount, not the input USD amount.

I think this bug already existed in production because we passed the input amount instead of the output amount.

mikasackermn replied:
fixed

mikasackermn commented:
> I think the value of realUsdValue should also be updated Because now when we show zero dollar amount, the tooltip is still showing the previous amount.
> 
> Please check if it needs to be updated elsewhere

fixed

nikaaru reviewed (approved):
thanks,
LGTM

Ikari-Shinji-re commented on widget/embedded/src/containers/Inputs/Inputs.tsx:
```diff
@@ -40,10 +41,14 @@ export function Inputs(props: PropTypes) {
   const fromTokenFormattedBalance =
     formatBalance(fromTokenBalance)?.amount ?? '0';
 
+  const inputUsdAmount =
+    !inputUsdValue || !fromToken ? new BigNumber(0) : inputUsdValue;
+  const outputUsdAmount =
+    !outputUsdValue || !toToken ? new BigNumber(0) : outputUsdValue;
```
I believe it is better to fix this issue in the store, rather than using derived state. We have two functions for resetting the state of the quote store that will be invoked in this situation. You can set `outputUsdValue` to `null` in `resetFromBlockchain`.

mikasackermn replied:
fixed
